nvdaHelper/vbufBackends/gecko_ia2/gecko_ia2.cpp

```diff
@@ -950,25 +950,20 @@ VBufStorage_fieldNode_t* GeckoVBufBackend_t::fillVBuf(
 
 		} else if (renderChildren && childCount > 0) {
 			// The object has no text, but we do want to render its children.
-			VARIANT* varChildren;
-			if(!(varChildren=(VARIANT*)malloc(sizeof(VARIANT)*childCount))) {
-				LOG_DEBUG(L"Error allocating varChildren memory");
-				return NULL;

```

Comment:
Here we return NULL earlier

Reply:
With the new approach we don't have to explicitly allocate memory, this is replaced by a `std::vector` inside the call to `getAccessibleChildren`. The `std::vector` may fail to allocate (if the `childCount` value was very large, or the system has run out of memory). We aren't explicitly checking for these cases, there isn't much we can do if the system is out of memory, but perhaps we should be checking for large values to protect against bugs (in a defensive programming sense)